packages/cli/src/eventbus/event-relay.service.ts

```diff
@@ -0,0 +1,18 @@
+import { EventEmitter } from 'node:events';
+import { Service } from 'typedi';
+import type { AuditEventArgs } from './audit.types';
+
+type EventArgs = AuditEventArgs; // also `TelemetryEventArgs` in future
```

Comment:
This kinda couples `EventRelay` to audit events and telemetry events, even tho just via the types and concepts. If we wanted to make this generic and fully decoupled, we could define a set of "domain" events. E.g. `UserCreatedEvent`, `UserSignedUpEvent`, etc. and `AuditEventRelay` would transform that into an `UserCreatedAuditEvent` and future Telemetry relay would transform that into `UserCreatedTelemetryEvent`. That way event producers also don't need to be aware of what consumers there are for the events. It does create more indirection and the event needs to contain all the data needed for the new events. I would be perhaps a bit more inclined towards this model. WDYT?

Reply:
Thank you for this idea! What I did right now was generalize the naming - next as we work on telemetry and start seeing the differences (or not?) between event payloads, we can create more fine-grained event types tailored to the consumer. Let me know if that's okay for now.

Reply:
Yeah this already makes things a lot better. Let's revisit if needed 👍